_code/InSituTree/R/runInSituTree.R

```diff
@@ -0,0 +1,183 @@
+    print(paste0("cth is of class ", class(cth)))
+    stop("Error: cth must be a list or vector of cell types.")
+  }
+
+  if (!is.numeric(neg)) {
+    stop("Error: neg must be a numeric vector.")
+  }
+
+  if (!is.character(name_for_new_annotation) ||
+        nchar(name_for_new_annotation) == 0) {
+    stop("Error: name_for_new_annotation must be a non-empty character string.")
+  }
+
+  if (!is.null(cohort) && !is.vector(cohort)) {
+    stop("Error: cohort must be a vector if provided.")
+  }
+
+  if (!is.null(excluded_genes) && !is.character(excluded_genes)) {
+    stop("Error: excluded_genes must be a character vector.")
+  }
+
+  if (!is.numeric(quantile_absolute_expression_difference_param) ||
+        quantile_absolute_expression_difference_param < 0 ||
+        quantile_absolute_expression_difference_param > 1) {
+    stop("Error: quantile_absolute_expression_difference_param must be
+           a numeric value between 0 and 1.")
+  }
+
+  if (!is.numeric(quantile_percent_expression_difference_param) ||
+        quantile_percent_expression_difference_param < 0 ||
+        quantile_percent_expression_difference_param > 1) {
+    stop("Error: quantile_percent_expression_difference_param must be
+           a numeric value between 0 and 1.")
+  }
+
+  out <- list()
+
+  # Make combined profile
+  prof <- collapseProfiles(cth = cth, full_profiles = full_profiles)
+
+  # annotate with InSituType
+  res <- supervisedSubcluster(
+    reference_profiles = prof,
+    x = x,
+    neg = neg,
+    cohort = cohort,
+    excluded_genes = excluded_genes,
+    quantile_absolute_expression_difference =
+      quantile_absolute_expression_difference_param,
+    quantile_percent_expression_difference =
+      quantile_percent_expression_difference_param
+  )
+  out[[name_for_new_annotation]] <- list()
+  out[[name_for_new_annotation]][["result"]] <- res
+  out[[name_for_new_annotation]][["subclusterings"]] <- list()
+  out[[name_for_new_annotation]][["name"]] <- name_for_new_annotation
+
+  # annotation next level down
+
+  for (i in names(cth)) {
+    if (i == "list" || i == name_for_new_annotation || i == "") {
+      next
+    }
+    message(i)
+    # check to make sure further subtyping is required.
+    # This is where cell lineages that are not present are dropped out.
+    celltypes_dropped <- !i %in% unique(res$clust)
+    if (celltypes_dropped) {
+      message(paste0("Warning: No cells found on which 
+                     to perform further fitting.  Dropping celltypes: ", i))
+    }
+
+    if (length(cth[[i]]) > 1 && !celltypes_dropped) {
+      message(paste0("Annotating  ", i, " cells..."))
+      selected_cells <- match(names(res$clust)[res$clust == i], rownames(x))
+      sub_res <- runInSituTree(
+        full_profiles = full_profiles,
+        cth = cth[[i]],
+        x = x[selected_cells, ],
+        neg = neg[selected_cells],
+        name_for_new_annotation = i,
+        cohort = cohort[selected_cells, ],
```

Comment:
It should be `cohort = cohort[selected_cells]` since it's a vector, not a data.frame.

Reply:
Good catch, changed.